### PR TITLE
Added API to delete a task

### PR DESCRIPTION
**Pull request by tanujvyas10**
Resolved Issue no #18

### CODE REVIEW

**Comment by tanujvyas10** [comment]
@devesh-verma Please review my work and please help in passing this checkings

File: app/routes/post.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ router.patch(
 // DELETE A TASK
 router.delete(
```
Comment (Rupeshiya) [inline app/routes/post.js]: @tanujvyas10 To delete the user should be authenticated so add the middleware to check if authenticated or not.
you can use the middleware which is already written in the middleware folder.

File: app/controllers/post.js [hunk 2]
```diff
@@ -26,6 +26,15 @@ module.exports = {
       }
     })
   },
+  delete: function (req, res, next) {
+    userModel.findByIdAndRemove(req.params.id, function (err) {
+      if (err) {
```
Comment (Rupeshiya) [inline app/controllers/post.js]: @tanujvyas10 if there is any error just throw it

Comment (tanujvyas10) [reply app/controllers/post.js]: Required changes have been made.Please review

**Comment by Rupeshiya** [comment]
@vaibhavdaren Why these code report checks are failed?

**Comment by tanujvyas10** [comment]
@devesh-verma @jaskirat2000 @vaibhavdaren Please review my PR

File: app/controllers/post.js (hunk 2 above)
Comment (devesh-verma) [reply app/controllers/post.js]: what are you trying to delete here?

Comment (devesh-verma) [reply app/controllers/post.js]: you are consuming User model here why?
Also, task will be deleted with the help of task_id

File: app/routes/post.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 const express = require('express')
 const router = express.Router()
 const userController = require('../controllers/post')
```
Comment (devesh-verma) [inline app/routes/post.js]: plz change the name. it should be `postController`

File: app/routes/post.js [hunk 4]
```diff
@@ -29,7 +29,8 @@ router.patch(
 // DELETE A TASK
 router.delete(
   '/:id',
-  userController.test
+  checkAuth,
+  userController.delete
```
Comment (devesh-verma) [inline app/routes/post.js]: same here

Comment (tanujvyas10) [reply app/routes/post.js]: ok I will update the changes soon

**Pull request closed without merging**